autoscale/src/main/java/com/sequenceiq/periscope/monitor/handler/ClusterStatusSyncHandler.java

```diff
@@ -44,42 +64,98 @@ public void onApplicationEvent(ClusterStatusSyncEvent event) {
 
-        updateClusterState(cluster, statusResponse, clusterAvailable);
+        updateClusterState(cluster, stackResponse, clusterAvailable);
     }
 
-    private void updateClusterState(Cluster cluster, StackStatusV4Response statusResponse, boolean clusterAvailable) {
-        if (DELETE_COMPLETED.equals(statusResponse.getStatus())) {
+    private void updateClusterState(Cluster cluster, StackV4Response stackResponse, boolean clusterAvailable) {
+        if (DELETE_COMPLETED.equals(stackResponse.getStatus())) {
             beforeDeleteCleanup(cluster);
             clusterService.removeById(cluster.getId());
-            LOGGER.info("Deleted cluster '{}', CB Stack Status '{}'.", cluster.getStackCrn(), statusResponse.getStatus());
+            LOGGER.info("Deleted cluster '{}', CB Stack Status '{}'.", cluster.getStackCrn(), stackResponse.getStatus());
         } else if (clusterAvailable && !RUNNING.equals(cluster.getState())) {
             clusterService.setState(cluster.getId(), ClusterState.RUNNING);
             LOGGER.info("Updated cluster '{}' to Running, CB Stack Status '{}', CB Cluster Status '{}'.",
-                    cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
+                    cluster.getStackCrn(), stackResponse.getStatus(), stackResponse.getCluster().getStatus());
         } else if (!clusterAvailable && RUNNING.equals(cluster.getState())) {
-            clusterService.setState(cluster.getId(), ClusterState.SUSPENDED);
-            LOGGER.info("Suspended cluster '{}', CB Stack Status '{}', CB Cluster Status '{}'",
-                    cluster.getStackCrn(), statusResponse.getStatus(), statusResponse.getClusterStatus());
+            clusterService.setState(cluster.getId(), SUSPENDED);
+            LOGGER.info("Suspended cluster '{}', CB Stack Status '{}', CB Cluster Status '{}'", cluster.getStackCrn(), stackResponse.getStatus(),
+                    stackResponse.getCluster().getStatus());
         } else if (RUNNING.equals(cluster.getState()) && (cluster.getMachineUserCrn() == null || cluster.getEnvironmentCrn() == null)) {
             populateEnvironmentAndMachineUserIfNotPresent(cluster);
         }
     }
 
+    private boolean determineClusterAvailability(Cluster cluster, StackV4Response stackResponse, DependentHostGroupsV4Response dependentHostGroupsResponse,
+            Set<String> policyHostGroups) {
+        if (!Boolean.TRUE.equals(cluster.isStopStartScalingEnabled()) || undefinedDependencyPresent(dependentHostGroupsResponse, policyHostGroups)) {
+            return Optional.ofNullable(stackResponse.getStatus()).map(Status::isAvailable).orElse(false);
+        }
+
+        boolean dependentHostsUnhealthy = policyHostGroups.stream().anyMatch(hg -> dependentHostsUnhealthy(dependentHostGroupsResponse, stackResponse, hg));
+
+        return !(stackDeletionInProgress(stackResponse) || stackModificationInProgress(stackResponse)
```

Comment:
Think this needs one more check - if the cluster is in various stages of being `stopped`, autoscaling should be disabled.
Do we need to think about clusters which are in the process of being created ? (Not sure if modification in progress will capture this)

Reply:
Updated the method name

Reply:
STOPPED still needs to be considered ? That is not an 'IN_PROGRESS' state.